File: iosApp/iosApp/Presentation/Main/Explore/ExploreAction.swift
//
//  ExploreAction.swift
//  iosApp
//
//  Created by 홍희표 on 2026/03/06.
//

import Foundation

enum ExploreAction {
    case queryChanged(String)
    case regionChanged(ExploreUiState.RegionFilter)
    case sortChanged(ExploreUiState.SortFilter)
    case tabChanged(ExploreUiState.TabType)
    case cafeTapped(id: String)
    case maidTapped(id: String)
    case refresh
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/explore/ExploreAction.kt
package com.hhp227.concafe.presentation.main.explore

sealed interface ExploreAction {
    data class QueryChanged(val query: String) : ExploreAction
    data class RegionChanged(val region: ExploreUiState.RegionFilter) : ExploreAction
    data class SortChanged(val sort: ExploreUiState.SortFilter) : ExploreAction
    data class TabChanged(val tab: ExploreUiState.TabType) : ExploreAction
    data class ClickCafe(val id: String) : ExploreAction
    data class ClickMaid(val id: String) : ExploreAction
    data object LoadMoreCafes : ExploreAction
    data object LoadMoreMaids : ExploreAction
    data object Refresh : ExploreAction
}
